code/controllers/subsystem/timer.dm

```diff
@@ -22,10 +22,48 @@ SUBSYSTEM_DEF(timer)
 
 	var/list/clienttime_timers = list() //special snowflake timers that run on fancy pansy "client time"
 
+	var/last_invoke_tick = 0
+	var/static/last_invoke_warning = 0
+	var/static/bucket_auto_reset = TRUE
+
 /datum/controller/subsystem/timer/stat_entry(msg)
 	..("B:[bucket_count] P:[length(processing)] H:[length(hashes)] C:[length(clienttime_timers)]")
 
 /datum/controller/subsystem/timer/fire(resumed = FALSE)
+	var/lit = last_invoke_tick
+	var/last_check = world.time - TIMER_NO_INVOKE_WARNING
+
+	if(!bucket_count)
+		last_invoke_tick = world.time
+
+	if(lit && lit < last_check && last_invoke_warning < last_check)
+		last_invoke_warning = world.time
+		var/msg = "No regular timers processed in the last [TIMER_NO_INVOKE_WARNING] ticks[bucket_auto_reset ? ", resetting buckets" : ""]!"
+		message_admins(msg)
+		WARNING(msg)
+		if(bucket_auto_reset)
+			bucket_resolution = 0
+		
+		log_world("Active timers at tick [world.time]:")
+		for(var/I in processing)
+			var/datum/timedevent/TE = I
+			msg = "Timer: [TE.id]: TTR: [TE.timeToRun], Flags: [TE.flags], "
```

Comment:
well, its a little late now, but processing only stores timers not in a bucket, ie far out timers.